### PR TITLE
hyperconnect 1.0.3 (new cask)

### PR DESCRIPTION
**Pull request by WarmEpoch**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by khipp** [comment]
@WarmEpoch You can fix style-related issues automatically by running `brew style --fix hyper-connect.rb`.

File: Casks/h/hyper-connect.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  version "1.0.3"
+  sha256 :no_check
+
+  url "https://mijia-ios-adhoc.cnbj1.mi-fds.com/mijia-ios-adhoc/hyperconnect/HyperConnect.dmg",
+      verified: "mijia-ios-adhoc.cnbj1.mi-fds.com/mijia-ios-adhoc/hyperconnect/"
+  name "Hyper Connect"
+  name "小米互联服务"
+  desc "Cross-device interconnection service for Xiaomi ecosystem"
+  homepage "https://hyperos.mi.com/continuity"
+
```
Comment (khipp) [inline Casks/h/hyper-connect.rb:11]:
```suggestion

  livecheck do
    url "https://cn-device.interconnect.miui.com/package/mac/checkUpdate?version=0.0.0"
    strategy :json do |json|
      json.dig("data", "versionInfo", "version")
    end
  end

```

We can add a `livecheck` to retrieve the current version using the URL from the in-app updater. You can find these feeds using [Proxyman](https://github.com/ProxymanApp/Proxyman).

File: Casks/h/hyper-connect.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+cask "hyper-connect" do
+  version "1.0.3"
+  sha256 :no_check
+
+  url "https://mijia-ios-adhoc.cnbj1.mi-fds.com/mijia-ios-adhoc/hyperconnect/HyperConnect.dmg",
+      verified: "mijia-ios-adhoc.cnbj1.mi-fds.com/mijia-ios-adhoc/hyperconnect/"
```
Comment (khipp) [inline Casks/h/hyper-connect.rb:6]:
```suggestion
  sha256 "c5ccba949f29f1fda54f9ebcf0f16a7c79e71a28732a50f4894406ef3a07727d"

  url "https://cdn.cnbj1.fds.api.mi-img.com/mijia-ios-adhoc/hyperconnect/HyperConnect-#{version}.dmg",
      verified: "mi-img.com/mijia-ios-adhoc/hyperconnect/"
```

We can add the versioned URL used by the in-app updater.

**Review by khipp — COMMENTED** [review]
Thank you for your first contribution to `homebrew-cask`, @WarmEpoch! 🎉

I left a few suggestions below. Please also squash your commits to keep the history cleaner. Thanks!

File: Casks/h/hyper-connect.rb (hunk 1 above)
Comment (WarmEpoch) [reply Casks/h/hyper-connect.rb:11]: Sorry, when rummaging in pagehome, I didn't find a livecheck available.

File: Casks/h/hyper-connect.rb (hunk 2 above)
Comment (WarmEpoch) [reply Casks/h/hyper-connect.rb:6]: There are no versions available for identification in PageHome.

File: Casks/h/hyper-connect.rb (hunk 1 above)
Comment (khipp) [reply Casks/h/hyper-connect.rb:11]: @WarmEpoch No problem, you can just commit my suggestions. I have tested them locally.

File: Casks/h/hyper-connect.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+cask "hyper-connect" do
+  version "1.0.3"
+  sha256 "c5ccba949f29f1fda54f9ebcf0f16a7c79e71a28732a50f4894406ef3a07727d"
+
+  url "https://cdn.cnbj1.fds.api.mi-img.com/mijia-ios-adhoc/hyperconnect/HyperConnect-#{version}.dmg",
+      verified: "mi-img.com/mijia-ios-adhoc/hyperconnect/"
+  name "Hyper Connect"
```
Comment (khipp) [inline Casks/h/hyper-connect.rb:7]:
```suggestion
  name "HyperConnect"
```

Based on the website, it appears that the official name is spelled without a space, i.e., 'HyperConnect'. We should rename the cask (and filename) to `hyperconnect` accordingly.

File: Casks/h/hyper-connect.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  name "小米互联服务"
+  desc "Cross-device interconnection service for Xiaomi ecosystem"
```
Comment (khipp) [inline Casks/h/hyper-connect.rb:9]:
```suggestion
  desc "Cross-device interconnection service for the Xiaomi ecosystem"
```

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @WarmEpoch! 🎉

**Review by p-linnane — APPROVED** [review]

**Pull request merged**